### ISSUE TITLE
[Feature] Check constraint checker

### ISSUE THREAD

**Issue by chaadow** [djezzzl/database_consistency#181]
Hi,
First, thanks for this wonderful gem.

I was wondering if adding a checker between "validates_numericality_of" and a CHECK constraint can be easily implemented.

I can try to look into it by looking into how the presence validator is implemented if you consider this valuable.

**Comment by djezzzl** [comment]
Hi @chaadow,

Sorry that it took so long for me to answer you! 

First, thank you for your kind words; I'm glad the gem is valuable. 

Could you please say if the `CHECK` constraint is somewhat standard? I just think what would be the way to compare what is defined in `CHECK` with what should be there.

The most effortless check would ensure some `CHECK` constraint on the field with `validates_numericality_of` validation. 

Would you find this helpful?

**Comment by chaadow** [comment]
Yes I believe it's quite [standard](https://www.w3schools.com/sql/sql_check.asp) 

Yes i'm thinking of 2-way checkers. One that would autocorrect by adding a migration like this : 
```ruby
class CreateBooks < ActiveRecord::Migration
  def change
    add_check_constraint :books, "price > 100", name: "price_check"
  end
end
```

and the other way around, where a check constraint exists on a column ( for numericality) but no `validates_numericality_of` in the model.
---

Another use case for a check constraint is for a NOT NULL : 
<img width="703" alt="image" src="https://user-images.githubusercontent.com/1371733/222963338-abc074c6-94db-4171-bbe6-f4821dbd6268.png">

Since it is functionally equivalent to the column constraint NOT NULL, it can avoid false positives for the `validates_presence_of`  checker.

**Comment by chaadow** [comment]
@djezzzl Thank you very much! 🙏🏼
